modules/apps/batch-planner/batch-planner-web/src/main/resources/META-INF/resources/js/import/ImportPreviewModal.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayButton from '@clayui/button';
+import ClayModal from '@clayui/modal';
+import ClayTable from '@clayui/table';
+import React from 'react';
+
+import CellPreview from './CellPreview';
+
+const ImportPreviewModal = ({
+	closeModal,
+	fieldsSelections,
+	fileContent,
+	handleEditCell,
+	setStartImport,
+}) => {
+	const fileFieldsToMap = fieldsSelections
+		? Object.keys(fieldsSelections)
+		: {};
+
+	return (
+		<>
+			<ClayModal.Header>
+				{Liferay.Language.get('preview')}
+			</ClayModal.Header>
+
+			<ClayModal.Body className="p-2">
+				<ClayTable borderless hover={false}>
+					<ClayTable.Head>
+						<ClayTable.Row>
+							{Array.from(fileFieldsToMap).map(
+								(element, index) => {
+									return (
+										<ClayTable.Cell headingCell key={index}>
+											{element.charAt(0).toUpperCase() +
+												element.slice(1)}
+										</ClayTable.Cell>
+									);
+								}
+							)}
+						</ClayTable.Row>
+					</ClayTable.Head>
+
+					<ClayTable.Body className="inline-scroller w-100">
+						{fileContent?.map((row, index) => {
+							return (
+								<ClayTable.Row key={index}>
```

Comment:
Using an array index as a `key` is an anti-pattern 

More info [here](https://robinpokorny.medium.com/index-as-a-key-is-an-anti-pattern-e0349aece318)